conn.go

```diff
@@ -512,6 +510,17 @@ func (conn *Conn) Signal(ch chan<- *Signal) {
 	conn.signalsLck.Unlock()
 }
 
+// RemoveSignal removes the given channel from the list of the registered channels.
+func (conn *Conn) RemoveSignal(ch chan<- *Signal) {
+	conn.signalsLck.Lock()
+	for i := len(conn.signals) - 1; i >= 0; i-- {
+		if ch == conn.signals[i] {
+			conn.signals = append(conn.signals[:i], conn.signals[i+1:]...)
```

Comment:
Correct me if I'm wrong, but I believe this may introduce a memory leak. While the slice is no longer referencing the channel in question, it's still contained within the backing array and is thus still technically reachable and will not be garbage collected. The solution is to nil the item in the slice before re-slicing.